src/mono/System.Private.CoreLib/System.Private.CoreLib.csproj

```diff
@@ -85,7 +85,11 @@
   </PropertyGroup>
   <PropertyGroup Condition="'$(TargetOS)' == 'Browser'">
     <PlatformTarget>AnyCPU</PlatformTarget>
-    <DefineConstants>$(DefineConstants);TARGET_WASM</DefineConstants>
+    <DefineConstants>$(DefineConstants);TARGET_BROWSER</DefineConstants>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetOS)' == 'WASI'">
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <DefineConstants>$(DefineConstants);TARGET_WASI</DefineConstants>
```

Comment:
I think we don't need the Browser/WASI PropertyGroups here since we already define these constants in  System.Private.CoreLib.Shared.projitems